x/ccv/provider/keeper/relay.go

```diff
@@ -376,6 +394,11 @@ func (k Keeper) HandleSlashPacket(ctx sdk.Context, chainID string, data ccv.Slas
 	}
 	k.slashingKeeper.JailUntil(ctx, providerConsAddr, jailTime)
 
+	// LastDowntimeValsetUpdateId increases monotonically
+	// all slash requests that are less than LastDowntimeValsetUpdateId will be dropped
+	// during slash request validation in OnRecvSlashPacket
+	k.SetLastDowntimeValsetUpdateId(ctx, chainID, data.ValsetUpdateId)
```

Comment:
This should only be set for downtime infractions. Also, we should set it on receiving a SlashPacket instead of on handling it.

Reply:
Not sure about the flow where you set it on receive.

Seems more logical to me to set it once the packet is actually processed, not when it was first seen.

Can you elaborate?

Reply:
https://github.com/cosmos/interchain-security/pull/619#pullrequestreview-1228226404

Reply:
If `LastDowntimeValsetUpdateId` is set on handle, then you could receive multiple packets for the same downtime infractions and all of them will be queued.

Reply:
Done